### PR TITLE
Using get_path functionality + global variables

### PR DESCRIPTION
**Pull request by Jad-yehya**

### CODE REVIEW

File: datasets/msl.py [hunk 1]
```diff
@@ -26,34 +26,36 @@ class Dataset(BaseDataset):
     name = "MSL"
 
     install_cmd = "conda"
-    requirements = ["pandas", "requests"]
+    requirements = ["pandas", "requests", "pathlib"]
```
Comment (tomMoral) [inline datasets/msl.py]: `pathlib` is part of the stdlib in python, no need to put it as a requirement

File: datasets/msl.py [hunk 2]
```diff
@@ -26,34 +26,36 @@ class Dataset(BaseDataset):
...
 
     parameters = {
         "debug": [False],
     }
 
     def get_data(self):
         # Adding get_data_path method soon
+        train_path = config.get_data_path(key="msl_train")
```
Comment (tomMoral) [inline datasets/msl.py]: just use a single key `msl` and put all files in the folder

File: datasets/psm.py [hunk 3]
```diff
@@ -22,34 +22,37 @@
 class Dataset(BaseDataset):
     name = "PSM"
     install_cmd = "conda"
-    requirements = ["pandas"]
+    requirements = ["pandas", "pathlib"]
```
Comment (tomMoral) [inline datasets/psm.py]: ditto

File: datasets/psm.py [hunk 4]
```diff
@@ -22,34 +22,37 @@
...
         "debug": [False],
     }
 
     def get_data(self):
         # Check if the data is already here
-        if not os.path.exists("data/PSM/PSM_train.csv"):
-            os.makedirs("data/PSM", exist_ok=True)
-            response = requests.get(url_xtrain)
-            with open("data/PSM/PSM_train.csv", "wb") as f:
+        train_path = config.get_data_path(key="psm_train")
```
Comment (tomMoral) [inline datasets/psm.py]: ditto

**Comment by Jad-yehya** [comment]
thanks @tomMoral  !

File: datasets/msl.py [hunk 5]
```diff
@@ -33,27 +33,22 @@ class Dataset(BaseDataset):
...
-        # Adding get_data_path method soon
-
+        path = config.get_data_path(key="MSL")
         # Check if the data is already here
-        if not os.path.exists("data/MSL/MSL_train.npy"):
-            os.makedirs("data/MSL", exist_ok=True)
-
-            response = requests.get(url_xtrain)
-            with open("data/MSL/MSL_train.npy", "wb") as f:
+        if not pathlib.Path.exists(path):
```
Comment (tomMoral) [inline datasets/msl.py:38]:
```suggestion
        if not path.exists():
```

File: datasets/msl.py [hunk 6]
```diff
@@ -33,27 +33,22 @@ class Dataset(BaseDataset):
...
+            response = requests.get(URL_XTRAIN)
+            with open(pathlib.Path(path) / "MSL_train.npy", "wb") as f:
```
Comment (tomMoral) [inline datasets/msl.py:40]:
```suggestion
            with path.open("wb") as f:
```

**Review by tomMoral — COMMENTED** [review]
i think `path` is directly a `Path` object

**Pull request merged**